File: Models/Predmet.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Models
{
    public class Predmet
    {
        [Key]
        public int ID { get; set; }
        [MaxLength(80)]
        [Required]
        public string Naziv { get; set; }

        [JsonIgnore]
        public ObjekatZaSmestanje Smesten{ get; set; }


    }
}

File: Client/Predmet.js
export class Predmet
{
    constructor(id,naziv,objekat)
    {
        this.ID=id;
        this.naziv=naziv;
        this.Objekat=objekat;
    }
    PremestiPredmet(objekat)
    {
        this.Objekat=objekat;
    }
}